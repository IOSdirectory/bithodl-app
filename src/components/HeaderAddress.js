import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import { ifIphoneX } from 'react-native-iphone-x-helper';

import BtcLabelContainer from '../containers/BtcLabelContainer';
import TimeLeftIcon from './TimeLeftIcon';

const UPDATE_INTERVAL = 60 * 1000; // 1 minute.

const styles = StyleSheet.create({
  view: {
    position: 'absolute',
    top: ifIphoneX(57, 33),
    right: 20,
    flexDirection: 'row',
    backgroundColor: 'rgba(255, 255, 255, 0.5)',
    padding: 5,
    paddingTop: 4,
    paddingBottom: 4,
    borderRadius: 40
  },
  icon: {
    width: 16,
    height: 16,
    borderRadius: 8,
    marginRight: 8
  },
  iconText: {
    fontSize: 7,
    backgroundColor: 'transparent'
  },
  amount: {
    fontSize: 12,
    fontWeight: '600',
    color: '#504F63',
    marginRight: 3
  }
});

export default class HeaderAddress extends Component {
  componentDidMount() {
    // Update component every minute.
    this._interval = setInterval(this.forceUpdate.bind(this), UPDATE_INTERVAL);
  }

  componentWillUnmount() {
    clearInterval(this._interval);
  }

  render() {
    const address = this.props.address;

    return (
      <View style={[styles.view, this.props.style]}>
        <TimeLeftIcon
          style={styles.icon}
          textStyle={styles.iconText}
          size={16}
          thickness={0.5}
          start={address.createdAt}
          end={address.unlockAt}
        />
        <BtcLabelContainer style={styles.amount} amount={address.amount} />
      </View>
    );
  }
}

HeaderAddress.propTypes = {
  style: PropTypes.any,
  address: PropTypes.object.isRequired
};
