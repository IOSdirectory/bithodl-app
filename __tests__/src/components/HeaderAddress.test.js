import React from 'react';
import renderer from 'react-test-renderer';
import HeaderAddress from '../../../src/components/HeaderAddress';

jest.mock('../../../src/containers/BtcLabelContainer', () => 'BtcLabelContainer');

describe('HeaderAddress', () => {
  it('renders correctly', () => {
    const fakeAddress = {
      id: '5012fed3-5d24-40f8-a99d-42bbfa7e9e56',
      hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
      script: 'OP_RETURN',
      amount: 2000000,
      createdAt: 1471651200,
      unlockAt: 1482192000
    };

    const tree = renderer.create(
      <HeaderAddress address={fakeAddress} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
