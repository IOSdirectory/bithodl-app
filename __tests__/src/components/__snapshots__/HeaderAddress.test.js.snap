// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderAddress renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "borderRadius": 40,
        "flexDirection": "row",
        "padding": 5,
        "paddingBottom": 4,
        "paddingTop": 4,
        "position": "absolute",
        "right": 20,
        "top": 33,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "height": 32,
            "justifyContent": "center",
            "width": 32,
          },
          Object {
            "backgroundColor": "#D9E5C3",
          },
        ],
        Object {
          "borderRadius": 8,
          "height": 16,
          "marginRight": 8,
          "width": 16,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": 0,
        }
      }
    >
      <View
        borderWidth={0}
        color="#7FA835"
        progress={1}
        size={16}
        thickness={0.5}
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Array [
            Object {
              "fontSize": 11,
              "fontWeight": "600",
            },
            Object {
              "color": "#7FA835",
            },
          ],
          Object {
            "backgroundColor": "transparent",
            "fontSize": 7,
          },
        ]
      }
    >
      âœ“
    </Text>
  </View>
  <BtcLabelContainer
    amount={2000000}
    style={
      Object {
        "color": "#504F63",
        "fontSize": 12,
        "fontWeight": "600",
        "marginRight": 3,
      }
    }
  />
</View>
`;
